create-budget/azuredeploy.json

```diff
@@ -0,0 +1,160 @@
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "budgetName": {
+      "type": "string",
+      "metadata": {
+        "description": "Budget name"
+      }
+    },
+    "amount": {
+      "type": "string",
+      "metadata": {
+        "description": "The total amount of cost or usage to track with the budget"
+      }
+    },
+    "budgetCategory": {
+      "type": "string",
+      "defaultValue": "Cost",
+      "allowedValues": [
+        "Cost",
+        "Usage"
+      ],
+      "metadata": {
+        "description": "The category of the budget, whether the budget tracks cost or usage."
+      }
+    },
+    "timeGrain": {
+      "type": "string",
+      "defaultValue": "Monthly",
+      "allowedValues": [
+        "Monthly",
+        "Quarterly",
+        "Annually"
+      ],
+      "metadata": {
+        "description": "The time covered by a budget. Tracking of the amount will be reset based on the time grain."
+      }
+    },
+    "startDate": {
+      "type": "string",
+      "defaultValue": "YYYY-MM-DD",
+      "metadata": {
+        "description": "The start date for the budget."
+      }
+    },
+    "endDate": {
+      "type": "string",
+      "defaultValue": "YYYY-MM-DD",
+      "metadata": {
+        "description": "The end date for the budget. If not provided, we default this to 10 years from the start date."
+      }
+    },
+    "operator": {
+      "type": "string",
+      "defaultValue": "GreaterThan",
+      "allowedValues": [
+        "EqualTo",
+        "GreaterThan",
+        "GreaterThanOrEqualTo"
+      ],
+      "metadata": {
+        "description": "The comparison operator."
+      }
+    },
+    "threshold": {
+      "type": "string",
+      "metadata": {
+        "description": "Threshold value associated with a notification. Notification is sent when the cost exceeded the threshold. It is always percent and has to be between 0 and 1000."
+      }
+    },
+    "contactEmails": {
+      "type": "array",
+      "defaultValue": [
+        "abc@contoso.com",
+        "xyz@contoso.com"
+      ],
+      "metadata": {
+        "description": "Email addresses to send the budget notification to when the threshold is exceeded."
+      }
+    },
+    "contactRoles": {
+      "type": "array",
+      "defaultValue": [
+        "Owner",
+        "Contributor",
+        "Reader"
+      ],
+      "metadata": {
+        "description": "Contact roles to send the budget notification to when the threshold is exceeded."
+      }
+    },
+    "contactGroups": {
+      "type": "array",
+      "defaultValue": [
+        "/subscriptions/{subscription-id}/resourceGroups/{resource-group-name}/providers/microsoft.insights/actionGroups/SampleActionGroup1"
+      ],
+      "metadata": {
+        "description": "Contact roles to send the budget notification to when the threshold is exceeded."
+      }
+    },
+    "resourcesFilter": {
+      "type": "array",
+      "defaultValue": [
```

Comment:
remove this

Reply:
Sure! I have removed the default value here.